src/hotspot/share/adlc/formssel.cpp

```diff
@@ -1533,7 +1535,7 @@ Predicate *InstructForm::build_predicate() {
         s += strlen(s);
       }
       // Add predicate to working buffer
-      sprintf(s,"/*%s*/(",(char*)i._key);
+      snprintf_checked(s, remaining_buflen(buf, s), "/*%s*/(",(char*)i._key);
```

Comment:
[pre-existing] the result from this snprintf could be used instead of strlen in the next line.

Reply:
I'm not very sure of why "+=" is used yet.  I would like to leave as it is.